compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -294,6 +295,29 @@ private void findTypeParam(Location loc, BType expType, BType actualType, Symbol
         }
         // Bound type is a structure. Visit recursively to find bound type.
         switch (expType.tag) {
+            case TypeTags.XML:
+                if (TypeTags.isXMLTypeTag(actualType.tag)) {
+                    switch (actualType.tag) {
+                        case TypeTags.XML:
+                            BType constraint = ((BXMLType) actualType).constraint;
+                            while (constraint.tag == TypeTags.XML) {
+                                actualType = constraint;
+                                constraint = ((BXMLType) actualType).constraint;
```

Comment:
Can't we directly do 
```suggestion
                                constraint = ((BXMLType) constraint).constraint;
```
instead?